### ISSUE TITLE
JSONParser does not handle JSON texts consisting of an array

### ISSUE THREAD

**Issue by fcami** [RedHatInsights/insights-core#2056]
As per https://tools.ietf.org/html/rfc7159.html#section-2 a JSON text can be many things including an array [] or an object {}.

The JSONParser implementation at:

https://github.com/RedHatInsights/insights-core/blob/8c633203ba72de49b70d0612e45398ed036197d4/insights/core/__init__.py#L717

incorrectly expects a JSON text to be an object (e.g. between {}) as it calls dict() on it.
See: https://docs.python.org/3/library/json.html#encoders-and-decoders

If the JSON text parsed is an array instead a backtrace occurs:

```
WARNING:insights.core.dr:Traceback (most recent call last):
  File "insights-core/insights/core/dr.py", line 954, in run
    result = DELEGATES[component].process(broker)
  File "insights-core/insights/core/dr.py", line 673, in process
    return self.invoke(broker)
  File "insights-core/insights/core/plugins.py", line 123, in invoke
    return self.component(dep_value)
  File "insights-core/insights/core/__init__.py", line 86, in __init__
    self._handle_content(context)
  File "insights-core/insights/core/__init__.py", line 89, in _handle_content
    self.parse_content(context.content)
  File "insights-core/insights/core/__init__.py", line 724, in parse_content
    self.doc = from_dict(self.data)
  File "insights-core/insights/configtree/__init__.py", line 369, in from_dict
    return Root(children=inner(dct))
  File "insights-core/insights/configtree/__init__.py", line 352, in inner
    for name, v in d.items():
AttributeError: 'list' object has no attribute 'items'
```

**Comment by fcami** [comment]
hi @bfahr @csams could you please have a look at the PR?

**Comment by miclark** [comment]
Assigning to @fcami since he seems to be working it.

**Comment by bfahr** [comment]
Fixed by #2057
